pandas/core/arrays/base.py

```diff
@@ -477,6 +486,16 @@ def take(self, indexer, allow_fill=True, fill_value=None):
            def take(self, indexer, allow_fill=True, fill_value=None):
                indexer = np.asarray(indexer)
                mask = indexer == -1
+
+               # take on empty array not handled as desired by numpy
+               if not len(self):
+                   # only valid if result is an all-missing array
+                   if mask.all():
+                       return type(self)([self._na_value] * len(indexer))
+                   else:
+                       raise IndexError(
+                           "cannot do a non-empty take from an empty array.")
```

Comment:
is this raise condition hit in a test?

Reply:
Yes, see the single test I added in this PR: it tests both the all missing case and this error